crates/ltk_texture/src/tex/encode.rs

```diff
@@ -155,8 +268,17 @@ pub fn encode_rgba_with_mipmaps(
 /// Encode RGBA8 data to BC1 format
 #[cfg(feature = "intel-tex")]
 fn encode_bc1(width: u32, height: u32, rgba_data: &[u8]) -> Result<Vec<u8>, EncodeError> {
+    let expected_len = width as usize * height as usize * 4;
+    if rgba_data.len() != expected_len {
+        return Err(EncodeError::InvalidPixelData);
+    }
+
```

Comment:
could be nice to make this whole process more modular down the line, to allow plugging in different dither impls or providing your own